admin/server/reports.go

```diff
@@ -283,7 +277,23 @@ func (s *Server) UnsubscribeReport(ctx context.Context, req *adminv1.Unsubscribe
 			return nil, status.Errorf(codes.InvalidArgument, "failed to find report token: %s", err.Error())
 		}
 
-		userEmail = reportTkn.RecipientEmail
+		if reportTkn.RecipientEmail == "" {
+			if req.Email != nil {
+				return nil, status.Error(codes.InvalidArgument, "anon token cannot be used for unsubscribing email recipients")
+			}
+			if req.SlackUser == nil {
+				return nil, status.Error(codes.InvalidArgument, "no slack user provided for unsubscribing")
+			}
+			userEmail = *req.SlackUser
+		} else {
+			userEmail = reportTkn.RecipientEmail
+			if req.Email == nil {
+				return nil, status.Error(codes.InvalidArgument, "no email provided for unsubscribing")
+			}
+			if strings.EqualFold(userEmail, *req.Email) {
+				return nil, status.Error(codes.InvalidArgument, "email does not match token")
+			}
```

Comment:
Should this be a NOT?

```suggestion
			if !strings.EqualFold(userEmail, *req.Email) {
				return nil, status.Error(codes.InvalidArgument, "email does not match token")
			}
```